internal/pkg/agent/application/upgrade/step_unpack.go

```diff
@@ -204,6 +215,61 @@ func getPackageMetadataFromZip(archivePath string) (packageMetadata, error) {
 	return getPackageMetadataFromZipReader(r, fileNamePrefix)
 }
 
+func skipFnFromZip(log *logger.Logger, r *zip.ReadCloser, detectedFlavor string, fileNamePrefix string, versionedHome string, registry map[string][]string) (install.SkipFn, error) {
+	if detectedFlavor == "" {
+		// no flavor don't skip anything
+		return func(relPath string) bool { return false }, nil
+	}
+
+	flavor, err := install.Flavor(detectedFlavor, "", registry)
+	if err != nil {
+		if errors.Is(err, install.ErrUnknownFlavor) {
+			// unknown flavor fallback to copy all
+			return func(relPath string) bool { return false }, nil
+		}
+		return nil, err
+	}
+	specsInFlavor, _ := install.SpecsForFlavor(flavor) // ignoring error flavor exists, it was loaded before
+
+	// fix versionedHome
+	versionedHome = strings.ReplaceAll(versionedHome, "\\", "/")
+	log.Errorf("using %q", versionedHome)
```

Comment:
Is this needed? It's not an error, right?

Reply:
leftover from previous design, thanks for pointing it out